packages/client/cli/src/service/cryptid/index.ts

```diff
@@ -100,21 +116,40 @@ export const airdrop = async (
   );
 };
 
+const signCryptidTransaction = (
+  cryptid: CryptidClient,
+  transaction: Transaction
+): Promise<Transaction[]> => {
+  if (cryptid.details.middlewares.length) {
+    return cryptid.proposeAndExecute(transaction);
+  } else {
+    return cryptid.directExecute(transaction).then((tx) => [tx]);
+  }
+};
+
+export const signAndSendCryptidTransaction = async (
```

Comment:
Should these things not rather live in core?

Reply:
potentially some kind of `sendAllInOrder` is a useful enough utility function to add to core. Not sure about the generalisation in signCryptidTransaction. If we use something similar in the wallet then maybe. But really we need something a little more complex, like the v1 signLarge, that can detect the size of a tx and decide if it needs splitting